optaplanner-core/pom.xml

```diff
@@ -134,6 +134,33 @@
           </ignoredUnusedDeclaredDependencies>
         </configuration>
       </plugin>
+
+      <plugin>
+        <artifactId>maven-resources-plugin</artifactId>
+        <executions>
+          <execution>
+            <id>add-annotation-processor-service</id>
+            <phase>prepare-package</phase>
+            <goals>
+              <goal>copy-resources</goal>
+            </goals>
```

Comment:
I am not a fan of complicating our build.
The solution could be worse than the problem.

Reply:
In an ideal world, the annotation processor + annotations would be in a different maven module, and optaplanner-core would use it as a dependency. But that would be a major refactor and split API into two places.

Reply:
How much time does it take to execute this goal? 
After looking into the scope of splitting the entire `api` package into a separate module, I am afraid we would have to keep this workaround in place.

Reply:
If the annotations are moved to the api module, the workaround can be removed and the file directly placed in optaplanner-core/src/main/resources. The goal itself take < 1 second to execute (it is a file copy)